### PR TITLE
Add additional TIL on reduce() method

### PR DESCRIPTION
**Pull request by danmarcus111**
I wrote an additional short TIL on some of the additional functionality of the reduce() method. I hope its suitable!

### CODE REVIEW

**Comment by lucymonie** [comment]
Hey @danmarcus111, I just noticed this was here. You may want to assign someone the task of checking over your TIL and merging it (or you could ask for someone to do it in the READMEs gitter room). Otherwise it might just sit here!

File: javascript/moreReduceMethod.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+var longest = arr.reduce(function(a,b){ return b.length > a.length ? b : a});
+```
+
+For each item of the array, the reduce method will check what was previously returned by it and compare it against the current array item.  If the current array item is bigger it will return that item to the next loop.  If not, it will return the original item to the next loop.
+
+### Using reduce() instead of forEach()
+
+The examples we have seen so far in this and the other TILs have only used the first two arguments in the callback function of the reduce() method.  However, this function can also receive the index of the current array object (this would be the third argument) and the array being traversed (this would be the fourth argument).
+
+In this respect, reduce() is very much like forEach() except it has an additional first argument (the memo).  When you are going to use either method, it may worth considering whether the other method may be more useful in the context.
```
Comment (Cleop) [inline javascript/moreReduceMethod.md]: 27 - When you are thinking of using either method, it's worth considering which method may be more useful in the context. Suggested rewording.

**Comment by Cleop** [comment]
Hey @danmarcus111. I've read through and there's just one small typo on line 27 which I gave a suggested rewording for. If you're happy to make that change then I'm happy to merge for you :-)

**Comment by marisid** [comment]
Hi @danmarcus111! It looks good that you referenced other TILs and added some more info on both `reduce()` and `forEach()` methods. May I suggest that you add an example when you compare functionality of `reduce()` and `forEach()` methods and maybe point out that `reduce()` returns a single value after executing the callback function whereas`forEach()` executes the callback function but returns undefined by default. Hope this makes sense - just a couple of suggestions :smile:

**Pull request merged**